packages/react-core/src/hooks/useDeprecationWarning.ts

```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+
+export type UseDeprecationWarning = (params: {
+  shouldWarn: boolean;
+  message: string;
+}) => void;
+
+/**
+ * Logs a deprecation warning message.
+ *
+ * @important Please use the React/React Native specific platform implementations.
+ * This version of the hook is a base implementation that the others extend from due
+ * to env differences between running in RN or the browser
+ */
+const useDeprecationWarning: UseDeprecationWarning = ({
+  shouldWarn,
+  message,
+}) => {
+  React.useEffect(() => {
+    if (shouldWarn) {
+      // eslint-disable-next-line no-console
+      console.warn(message);
```

Comment:
Could we use Logger from `aws-amplify` library and not have to eslint-disable? Thinking it should have the same outcome.

Reply:
Discussed offline